### PR TITLE
UnHardWire Plugins for HLT 75X

### PR DESCRIPTION
**Pull request by Martin-Grunewald**
This PR also contains #7781 #8382 and content of #8379 (8379 itself conflicted with 7781),
plus the required HLT modifications.

### CODE REVIEW

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
@mtosi @VinInn 

The error message is
```
# Conditions read from  CMS_CONDITIONS  via FrontierProd 
19-Mar-2015 04:55:50 CET  Initiating request to open file root://eoscms//eos/cms/store/hidata/HIRun2011/HIMinBiasUPC/RAW/v1/000/182/124/0011F55A-8F13-E111-A987-003048F118C6.root?svcClass=default
%MSG-w XrdAdaptor:  file_open 19-Mar-2015 04:55:50 CET pre-events
Data is served from cern.ch instead of original site eoscms
%MSG
19-Mar-2015 04:55:51 CET  Successfully opened file root://eoscms//eos/cms/store/hidata/HIRun2011/HIMinBiasUPC/RAW/v1/000/182/124/0011F55A-8F13-E111-A987-003048F118C6.root?svcClass=default
%MSG-w MultiTRackSelector:  MultiTrackSelector:hiInitialStepSelector@ctor  19-Mar-2015 04:55:57 CET pre-events
you are executing buggy code, if intentional please help to fix it
%MSG
%MSG-w MultiTRackSelector:   MultiTrackSelector:hiDetachedTripletStepSelector@ctor 19-Mar-2015 04:55:57 CET  pre-events
you are executing buggy code, if intentional please help to fix it
%MSG
%MSG-w MultiTRackSelector:  MultiTrackSelector:hiLowPtTripletStepSelector@ctor  19-Mar-2015 04:55:57 CET pre-events
you are executing buggy code, if intentional please help to fix it
%MSG
%MSG-w MultiTRackSelector:  MultiTrackSelector:hiPixelPairStepSelector@ctor  19-Mar-2015 04:55:57 CET pre-events
you are executing buggy code, if intentional please help to fix it
%MSG
Begin processing the 1st record. Run 182124, Event 1893722, LumiSection 40 at 19-Mar-2015 04:57:08.115 CET
cmsRun: /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-18-2300/src/RecoTracker/TkSeedGenerator/plugins/SeedGeneratorFromProtoTracksEDProducer.cc:138: virtual void SeedGeneratorFromProtoTracksEDProducer::produce(edm::Event&, const edm::EventSetup&): Assertion `hits.size()<4' failed.
```

Please have a look: it seems you added this assert - since it wasn't there before maybe it
should just be taken out again.

**Comment by Martin-Grunewald** [comment]
@mtosi @VinInn @makortel 
In fact, Matti had removed this assert explicitly here:
https://github.com/cms-tsg-storm/cmssw/commit/2b246348f1aced0bee687c78a83bb200db7a2e8

File: RecoTracker/TkSeedGenerator/plugins/SeedGeneratorFromProtoTracksEDProducer.cc [hunk 1]
```diff
@@ -122,10 +135,13 @@ void SeedGeneratorFromProtoTracksEDProducer::produce(edm::Event& ev, const edm::
         if(refHit->isValid()) hits.push_back((Hit)&(*refHit));
       }
       sort(hits.begin(), hits.end(), HitLessByRadius());
+      assert(hits.size()<4);
```
Comment (slava77) [inline RecoTracker/TkSeedGenerator/plugins/SeedGeneratorFromProtoTracksEDProducer.cc]: the merge didn't go right, apparently.
Was this a conflict resolved the wrong way?

File: RecoTracker/TkSeedGenerator/plugins/SeedGeneratorFromProtoTracksEDProducer.cc [hunk 2]
```diff
@@ -24,6 +24,7 @@
 
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 #include <vector>
+#include <cassert>
```
Comment (slava77) [inline RecoTracker/TkSeedGenerator/plugins/SeedGeneratorFromProtoTracksEDProducer.cc]: same part as https://github.com/cms-sw/cmssw/pull/8393/files#r26731809
both are in 75X for ~two weeks, as of #7798

Comment (Martin-Grunewald) [inline RecoTracker/TkSeedGenerator/plugins/SeedGeneratorFromProtoTracksEDProducer.cc]: Can I remove the assert - or is this like hiding a bug ?

Comment (makortel) [inline RecoTracker/TkSeedGenerator/plugins/SeedGeneratorFromProtoTracksEDProducer.cc]: My take is that the assert and `#include <cassert>` should be removed (as in https://github.com/makortel/cmssw/commit/2b246348f1aced0bee687c78a83bb200db7a2e8b). There should be no need to add them back. (Was the file in this branch perhaps copied from 74X?)

Comment (Martin-Grunewald) [inline RecoTracker/TkSeedGenerator/plugins/SeedGeneratorFromProtoTracksEDProducer.cc]: I think so - ok, will remove for 74X and 75X versions of this PR.
Thanks!

Comment (makortel) [inline RecoTracker/TkSeedGenerator/plugins/SeedGeneratorFromProtoTracksEDProducer.cc]: For 74X there is no strict need to remove the assert, as the code causing it to fire is only in 75X. On the other hand, removing it from 74X should not cause any problems either.

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

**Comment by slava77** [comment]
now that the large forward port was merged (#8357) this is no longer mergeable

**Comment by Martin-Grunewald** [comment]
Rebased to 75X

**Comment by Martin-Grunewald** [comment]
please test

**Comment by cvuosalo** [comment]
+1

For #8393 ac42fb45d35515ddd472c7fc9fe2051416e8287e

Fixes for the HLT configuration. This PR is the same as #8390 for 74X, which has already been approved by Reco.

The code changes are satisfactory, and Jenkins test [results] (https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8393/3642/summary.html) against baseline CMSSW_7_5_X_2015-03-18-2300 show no significant differences, except for minor differences in some HLT plots (see #8390) that are expected as a result of this PR.

**Comment by Martin-Grunewald** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**